depth_first_traversal/DepthFirstTraversal.java

```diff
@@ -0,0 +1,60 @@
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.Set;
+
+public class DepthFirstTraversal {
+
+    public static int[] arr;
+    public static int k = 0;
+    // Array  of lists for Adjacency List Representation
+    public static LinkedList<Integer>[] adj;    
+    
+    //Function to add an edge into the DepthFirstTraversal
+    private static void addEdge(int v, int w) {
+        adj[v].add(w);  
+    }
+
+    // A function used by DFS
+    private static void depthFirstTraversal(int v, Set visited) {
+        // Mark the current node as visited 
+        visited.add(v);
+        arr[k] = v;
+        k++;        
+        Iterator<Integer> i = adj[v].listIterator();
+        while (i.hasNext()) {
+            int n = i.next();
+            if (!visited.contains(n)) {
+                depthFirstTraversal(n, visited);
+            }
+        }
+    }
+
+    public static void dFS(int v) {        
+        // false by default in java)
+        Set<Integer> visited = new HashSet();
+        // Call the recursive helper function to print DFS traversal
+        depthFirstTraversal(v, visited);
+    }
+
+    public static void main(String[] args) {
+    	int n = 4;
+        adj = new LinkedList[n];
```

Comment:
This is undesirable because how nodes are stored is a problem of the DFS class. 
User should be able to use high level methods to do these tasks. Eg > `initEdges(n)`